modules/govuk/manifests/node/s_logs_elasticsearch.pp

```diff
@@ -35,6 +35,24 @@
     ],
   }
 
+  case $govuk_elasticsearch::version {
+    /^0.9/: { $cloud_aws_version = '1.16.0' }
+    /^1.0/: { $cloud_aws_version = '2.0.0' }
+    /^1.1/: { $cloud_aws_version = '2.1.1' }
+    /^1.2/: { $cloud_aws_version = '2.2.0' }
+    /^1.3/: { $cloud_aws_version = '2.3.0' }
+    /^1.4/: { $cloud_aws_version = '2.4.2' }
+    /^1.5/: { $cloud_aws_version = '2.5.1' }
+    /^1.6/: { $cloud_aws_version = '2.6.1' }
+    /^1.7/: { $cloud_aws_version = '2.7.1' }
+    default: { $cloud_aws_version = '2.7.1' }
+  }
```

Comment:
I think that's true, but until we reuse this code it's inside `govuk::node::s_logs_elasticsearch` and references versions of Elasticsearch that will never be running on those machines which might be confusing.

I understand that the documentation says that the plugin must be upgraded whenever Elasticsearch is, but the reality is that people forget things like that so we should write code which guards against those cases and fails early in the Puppet run to prevent issues.